docs/examples/01_current_injection.py

```diff
@@ -1,502 +1,694 @@
-# # curve between firing rate and input current:
-
-# fig, ax = plt.subplots(1, 1, figsize=(7, 5))
-# ax.plot(nap.compute_1d_tuning_curves(spikes[[1]], binned_current, 15), 'k', label="Observed")
-# ax.plot(nap.compute_1d_tuning_curves_continuous(pred_fr, np.squeeze(binned_current), 15), 'r--', label="Predicted")
-# ax.legend()
-# ax.set(ylabel="Firing rate [Hz]", xlabel="Input current [pA]")
-
-# # %%
-# #
-# # Predicted firing rate is a smooth exponential of the input current, while
-# # observed is not just noisier, but falls off rapidly at high levels of current.
-# #
-# # We can also recover this tuning directly from the GLM object:
-
-# print(glm.coef_.shape, glm.intercept_.shape)
-# current = np.linspace(0, 140)
-# fr = np.exp(glm.coef_ * current + glm.intercept_).squeeze()
-# # the firing rate calcualted above is in spikes per bin, so convert that to Hz
-# fr /= bin_size
-# ax.plot(nap.compute_1d_tuning_curves(spikes[[1]], binned_current, 15), 'k', label="Observed")
-# ax.plot(nap.compute_1d_tuning_curves_continuous(pred_fr, np.squeeze(binned_current), 15), 'r--', label="Predicted")
-# ax.set(ylabel="Firing rate [Hz]", xlabel="Input current [pA]")
-# ax.plot(current, fr, 'r', label="Predicted from GLM directly")
-# ax.legend()
-# fig
-# # %%
-# # Now:
-# #
-# # - compare number of spikes within each current interval
-# #
-# # - plot each current interval, show how rate is linear with input and so misses some effects
-# #
-# # - show tuning curves, point out that they differ at high input levels
-
-
-# # point out this is two parameters and is only using input at time t, say we
-# # can get a little fancier by adding history. try with and without bases?
-# #
-# # then add self-excitatin
+# %%
+#
+# In addition to making that mismatch discussed earlier a little more obvious,
+# this tuning curve comparison also highlights that this model thinks the
+# firing rate will continue to grow as the injected current increases, which is
+# not reflected in the data.
+#
+# Viewing this plot also makes it clear that the model's tuning curve is an
+# exponential. We already knew that! That's what it means to be a LNP model of
+# a single input. But it's nice to see it made explicit.
+#
+# ### Finishing up
+#
+# There are a handful of other operations you might like to do with the GLM.
+# First, you might be wondering how to simulate spikes -- the GLM is a LNP
+# model, but the firing rate is just the output of *LN*, its first two steps.
+# The firing rate is just the mean of a Poisson process, so we can pass it to
+# `jax.random.poisson`:
+
+spikes = jax.random.poisson(jax.random.PRNGKey(0), predicted_fr.values)
+
+# %%
+#
+# Note that this is not actually that informative and, in general, it is
+# recommended that you focus on firing rates when interpreting your model. In
+# particular, if your GLM includes auto-regressive inputs (e.g., neurons are
+# connected to themselves or each other), simulate can behave poorly, see
+# **XXX** for details.
```

Comment:
Maybe this could just be put under "exercises"?

Reply:
why so?